src/main/java/org/truffleruby/interop/RubyStringToJavaStringNode.java

```diff
@@ -27,16 +27,12 @@
 import org.truffleruby.language.control.RaiseException;
 import org.truffleruby.language.dispatch.CallDispatchHeadNode;
 
-/**
- * Take a Symbol or some object accepting #to_str
- * and convert it to a Java String.
- */
 @ImportStatic({ StringCachingGuards.class, StringOperations.class })
 @NodeChild(value = "value", type = RubyNode.class)
-public abstract class NameToJavaStringNode extends RubyNode {
+public abstract class RubyStringToJavaStringNode extends RubyNode {
```

Comment:
This seems a bad name given that it also accepts Symbol (and it must!).
i.e. This is often used for variable/method names and identifiers.

Reply:
It's often used for that, but not always, so I thought the old name was more confusing. I could call it `RubyStringOrSymbolToJavaStringNode`, but I thought that was getting silly.

Reply:
I think the old name was much less confusing :)
If you want something that only accepts Ruby String (+ maybe Symbol) but not `#to_str`, it should be another node.

Reply:
I could call it `InteropToJavaStringNode` - it does the Java string conversion, as needed for interop.

Reply:
...although it also used where we want a Java identifier in a couple of cases in `ModulesNodes` and `KernelNodes` I believe.

Reply:
Yes, actually I think most usages are for identifiers, and that's why I think the rename is bad.
Let's keep the original and replace interop usages with a new node, such as RubyToJavaStringNode, which can currently delegate to NameToJavaStringNode, but might have slightly different semantics. Then we have good names for both kinds of usages.

Reply:
That seems backwards to me - why would you want a node that does nothing to do with names to be called `NameToJavaStringNode`?

Reply:
Are `:foo` and `"foo"` when referring to a method or variable not "names"?
I'd be fine with IdentifierToJavaStringNode if that's clearer.

Reply:
Yeah, in that one use case. Say I have a JS template and I run it passing Ruby strings as data then they're not names or identifies or anything like that, they're just string data.

Reply:
I mean to keep using `NameToJavaStringNode` for things like `private`, `attr_reader`, and other ModuleNodes using it and friends.

And for interop, something like `RubyToJavaStringNode` or `InteropToJavaStringNode`.
Maybe those interop cases should simply use RubyToForeignNode.

Reply:
Ok I'll do a commit for this.